dotnet/src/Connectors/Connectors.Memory.AzureCognitiveSearch/AzureCognitiveSearchMemoryStore.cs

```diff
@@ -169,6 +169,9 @@ public async IAsyncEnumerable<string> UpsertBatchAsync(string collectionName, IE
         bool withEmbeddings = false,
         [EnumeratorCancellation] CancellationToken cancellationToken = default)
     {
+        // Cosine similarity range: -1 .. +1
+        minRelevanceScore = Math.Max(-1, Math.Min(1, minRelevanceScore));
```

Comment:
Move the line down closer to the place it's used to improve readability and decrease cognitive load.

Reply:
this is validation logic, it's common practice checking/cleaning when entering a method before any other logic takes place